File: src/main/java/kitchenpos/menu/tobe/application/MenuGroupService.java
package kitchenpos.menu.tobe.application;

import kitchenpos.menu.tobe.domain.menugroup.MenuGroup;
import kitchenpos.menu.tobe.domain.menugroup.MenuGroupRepository;
import org.springframework.stereotype.Service;
import org.springframework.transaction.annotation.Transactional;

import java.util.List;

@Service
public class MenuGroupService {
    private final MenuGroupRepository menuGroupRepository;

    public MenuGroupService(MenuGroupRepository menuGroupRepository) {
        this.menuGroupRepository = menuGroupRepository;
    }

    @Transactional
    public MenuGroup create(final MenuGroup request) {
        final MenuGroup menuGroup = MenuGroup.of(request.getNameValue());
        return menuGroupRepository.save(menuGroup);
    }

    @Transactional(readOnly = true)
    public List<MenuGroup> findAll() {
        return menuGroupRepository.findAll();
    }

}

File: src/main/kotlin/kitchenpos/menu/tobe/application/MenuGroupService.kt
package kitchenpos.menu.tobe.application

import kitchenpos.menu.tobe.application.dto.CreateMenuGroupReq
import kitchenpos.menu.tobe.application.dto.MenuGroupResp
import kitchenpos.menu.tobe.domain.MenuGroup
import kitchenpos.menu.tobe.domain.MenuGroupRepository
import org.springframework.stereotype.Service
import org.springframework.transaction.annotation.Transactional

@Service(value = "tobeMenuGroupService")
class MenuGroupService(
    private val menuGroupRepository: MenuGroupRepository,
) {

    @Transactional
    fun create(request: CreateMenuGroupReq): MenuGroupResp {
        val menuGroup = MenuGroup(
            name = request.name
        )
        menuGroupRepository.save(menuGroup)
        return MenuGroupResp.of(menuGroup)
    }

    fun findAll(): List<MenuGroupResp> {
        return menuGroupRepository.findAll()
            .map(MenuGroupResp::of)
    }
}
